From 97b353cc0469f6067f53117a9557d6750bd00dc2 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Wed, 13 Feb 2013 10:57:24 +0100
Subject: [PATCH 2/5] Resolves: rhbz#878253 crash when thin pool/volume is
 encountered

Thin pools/volumes are not supported by they are properly ignored.
---
 src/lvm_model.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/lvm_model.py b/src/lvm_model.py
index a4342c4..45c8b54 100644
--- a/src/lvm_model.py
+++ b/src/lvm_model.py
@@ -484,6 +484,11 @@ class lvm_model:
       lv_size = int(words[LV_SIZE_IDX]) / extent_size
       seg_size = int(words[LV_SEG_SIZE_IDX]) / extent_size
       devices = words[LV_DEVICES_IDX]
+
+      if attrs[0] == 't' or attrs[0] == 'V':
+        # thin pools/volumes are not supported
+        continue
+
       if LVS_HAS_MIRROR_OPTIONS:
         mirror_log = words[LV_MIRROR_LOG_IDX].strip()
       
-- 
2.1.4

